tests/test_jj_merge.lua

```
@@ -0,0 +1,373 @@
+    table.insert(results.tests, {name = name, status = "FAIL", error = tostring(err)})
+    print(string.format("✗ %s: %s", name, err))
+  end
+end
+
+local function assert_eq(actual, expected, msg)
+  if actual ~= expected then
+    error(string.format("%s\nExpected: %s\nActual: %s", msg or "Assertion failed", tostring(expected), tostring(actual)))
+  end
+end
+
+local function assert_not_nil(value, msg)
```

Luacheck (1.2.0)

[warning] unused function 'assert_not_nil'

(W211)